src/app/actions.ts

```diff
@@ -34,12 +23,42 @@ const fuseOptions = {
   includeScore: true,
   keys: ["name"],
 };
-const fuse = new Fuse<FlattenedRegion>(flattenedRegionCodes, fuseOptions);
+const regionsFuseIndex = new Fuse<FlattenedRegion>(
+  flattenedRegionCodes,
+  fuseOptions,
+);
 
 export async function findMatchingRegions(query: string) {
-  const regions = fuse.search(query);
+  const regions = regionsFuseIndex.search(query);
   if (!regions || regions.length === 0) {
     throw new Error(`Could not find any regions matching ${query}`);
   }
   return regions.slice(0, 10);
 }
+
+export type Species = {
+  sciName: string;
+  comName: string;
+  speciesCode: string;
+  category: string;
+  taxonOrder: number;
+  bandingCodes: string[];
+  comNameCodes: string[];
+  sciNameCodes: string[];
+  order: string;
+  familyCode: string;
+  familyComName: string;
+  familySciName: string;
+};
+
+export type SpeciesGetResponse = Array<{
+  item: Species;
+}>;
```

Comment:
This is re-exporting the Fuse type but with our own internal type name. It's an extension of the principles of abstraction we were talking about before. 

This means that if the internal implementation changes, all we need to do is make sure that this returns the same type and everything else will continue to work.

Reply:
Exactly that! We might have to do a little data manipulation to match this shape, but as long as we do then the rest of the code can stay exactly the same.